opennav_docking/src/navigator.cpp

```diff
@@ -66,19 +70,36 @@ void Navigator::goToPose(
       future_goal_handle, 2s) == rclcpp::FutureReturnCode::SUCCESS)
   {
     auto future_result = nav_to_pose_client_->async_get_result(future_goal_handle.get());
-    if (executor_.spin_until_future_complete(
-        future_result, timeout) == rclcpp::FutureReturnCode::SUCCESS)
-    {
-      auto result = future_result.get();
-      if (result.code == rclcpp_action::ResultCode::SUCCEEDED && result.result->error_code == 0) {
-        return;  // Success!
+
+    while (rclcpp::ok()) {
+      if (isPreempted()) {
+        nav_to_pose_client_->async_cancel_goal(future_goal_handle.get());
+        throw opennav_docking_core::Preempted("Navigation request to staging pose preempted.");
```

Comment:
How about fail to stage here instead of the new type? If its the `checkAndWarnIfCancelled` type, then that is cancelation and we don't use the type anyway since its user requested. If its a preemption, then the return error code is not all that interesting because we know that the new goal came through to execute. I think `FailedToStage` can cover it sufficiently here too -- but I'm open to push back on that